Sources/XRPLSwift-rest/Client/XrplClient.swift

```diff
@@ -0,0 +1,514 @@
+            throw err
+        }
+    }
+    
+    // swiftlint:disable:next identifier_name
+    func request(_ r: RipplePathFindRequest) async throws -> RipplePathFindResponse {
+        do {
+            let res: RipplePathFindResponse = try await request<RipplePathFindRequest>(rdict: r)
+            return res
+        } catch let err{
+            throw err
+        }
+    }
+    
+    // swiftlint:disable:next identifier_name
+    func request(_ r: ServerInfoRequest) async throws -> ServerInfoResponse {
+        do {
+            let res: ServerInfoResponse = try await request<ServerInfoRequest>(rdict: r)
+            return res
+        } catch let err{
+            throw err
+        }
+    }
+    
+    // swiftlint:disable:next identifier_name
+    func request(_ r: ServerStateRequest) async throws -> ServerStateResponse {
+        do {
+            let res: ServerStateResponse = try await request<ServerStateRequest>(rdict: r)
+            return res
+        } catch let err{
+            throw err
+        }
+    }
+    
+//    // swiftlint:disable:next identifier_name
+//    func request(_ r: SubmitRequest) async -> EventLoopFuture<Any> {
+//        return await request(r: r)
+//    }
+//    // swiftlint:disable:next identifier_name
+//    func request(_ r: SubmitMultisignedRequest) async -> EventLoopFuture<Any> {
+//        return await request(r: r)
+//    }
+//    //    // swiftlint:disable:next identifier_name
+//    //    func request(_ r: SubscribeRequest) async -> EventLoopFuture<Any> {
+//    //        return await request(r: r)
+//    //    }
+//    //    // swiftlint:disable:next identifier_name
+//    //    func request(_ r: UnsubscribeRequest) async -> EventLoopFuture<Any> {
+//    //        return await request(r: r)
+//    //    }
+    
+    // swiftlint:disable:next identifier_name
+    func request(_ r: TransactionEntryRequest) async throws -> TransactionEntryResponse {
+        do {
+            let res: TransactionEntryResponse = try await request<TransactionEntryRequest>(rdict: r)
+            return res
+        } catch let err{
+            throw err
+        }
+    }
+//    // swiftlint:disable:next identifier_name
+//    func request(_ r: TxRequest) async -> EventLoopFuture<Any> {
+//        return await request(r: r)
+//    }
+    // swiftlint:disable:next identifier_name
+
+    public func request<T: Request, R: Codable>(rdict: T) async throws -> R {
+        let req = BaseRequest<T>.init(method: T.getMethod(), params: [rdict])
+        guard let data = try JSONSerialization.jsonObject(with: req.jsonData()) as? APIParams else {
+            throw http.APIError.cannotParseRequest
+        }
+        
+        do {
+            let res: R = try await http.request(
+                httpMethod: http.HttpMethod.post,
+                url: URL.init(string: self.host)!,
+                params: data
+            )!
+            
+            return res
+        } catch let err {
+            throw err
+        }
```

Comment:
Rather than unwrapping, it might be better to maintain type safety and handle values explicitly using optional binding.

```suggestion
        guard let url = URL.init(string: self.host) else {
            throw http.APIError.invalidURL
        }

        do {
            guard let res: R = try await http.request(
                httpMethod: http.HttpMethod.post,
                url: url,
                params: data
            ) else{
                // Unexpected data format
                throw http.APIError.cannotParseRequest
            }
            
            return res;
        } catch let err {
            throw err
        }
```

Reply:
Thx to the commit [2a30e8a](https://github.com/girin-app/XRPLSwift-rest/pull/1/commits/2a30e8a7b675a2f8681508c68835e91fc4fffee7), the snippet is simplified as [c931798](https://github.com/girin-app/XRPLSwift-rest/pull/3/commits/c931798036a541e6e67dc485bb45b4d48d79ae4a)